File: backend/agenciaViajes.Application/Domain/Entities/Provider.cs
namespace agenciaViajes.Application.Domain.Entities
{
    public class Provider
    {
        public int Id { get; set; }
        public string Name { get; set; } = default!;
        public string Acronym { get; set; } = default!;
        public string Email { get; set; } = default!;
        public string Phone { get; set; } = default!;
        public string ProviderContactName { get; set; } = default!;
        public decimal? DepositPercentage { get; set; }
        public int? FinalPaymentDaysBefore { get; set; }
        public decimal? ProfitPercentage { get; set; }
        public bool Active { get; set; } = true;

    }
}

File: frontend/src/types/provider.ts
/**
 * Provider Types
 * Basado en Provider.cs del backend
 */

export interface Provider {
  id: number;
  name: string;
  acronym: string;
  email: string;
  phone: string;
  providerContactName: string;
  depositPercentage?: number; // Porcentaje de anticipo requerido
  finalPaymentDaysBefore?: number; // Días antes del viaje para liquidación final
  active: boolean;
}

export interface CreateProviderRequest {
  name: string;
  acronym: string;
  email: string;
  phone: string;
  providerContactName: string;
  depositPercentage?: number;
  finalPaymentDaysBefore?: number;
}

export interface UpdateProviderRequest {
  name: string;
  acronym: string;
  email: string;
  phone: string;
  providerContactName: string;
  depositPercentage?: number;
  finalPaymentDaysBefore?: number;
  active: boolean;
}
